### PR TITLE
Oversea IP Update

### PR DESCRIPTION
**Pull request by arition**
网易上有些冷门的歌分布在不同的缓存服务器上，本来应该是由DNS指定服务器IP的，但使用固定IP的话就会失败。

这个PR里增加了：
多选IP的窗口
IP列表自动获取ping
播放失败时自动切换IP
补充了一部分可用的缓存服务器IP

### CODE REVIEW

**Comment by EraserKing** [comment]
Hmmm... This is quite interesting. I'll merge it but will take a closer look before I publish.

File: CloudMusicGear/Proxy.cs [hunk 1]
```diff
@@ -211,7 +211,21 @@ private static void OnResponse(Session s)
...
+                    try
+                    {
+                        ipIndex = Config.IpAddressList.IndexOf(Config.IpAddress) + 1;
+                        if (ipIndex == Config.IpAddressList.Count) ipIndex = 0;
+                    }
+                    catch
+                    {
+                        if (Config.IpAddressList.Count > 0) ipIndex = 0;
+                    }
+                    if (ipIndex != null) Config.IpAddress = Config.IpAddressList[ipIndex.Value];
```
Comment (EraserKing) [inline CloudMusicGear/Proxy.cs:227]: I found... The piece of code here just redirects to next IP once, and it doesn't do anything to solve current 404 error.

For example, you selected three IPs: A, B and C. Current IP in use is A. The song doesn't exist on server A, so in this response, it still return 404 to the app (it doesn't do anything with the failure on hand), and next time it requests another song, it uses server B. In short words, it cycles to the next server for next song for current error.

Am I right?

**Comment by EraserKing** [comment]
You mean 404 error for "本来应该是由DNS指定服务器IP的，但使用固定IP的话就会失败", right?

**Comment by arition** [comment]
网易云音乐会对失败的请求重复请求4次，所以请求一次失败第二次请求就会用的新的IP

**Comment by EraserKing** [comment]
OK. I need a new client :)
2.0.2 did 5 attempts but 2.0.1 just once.

**Comment by arition** [comment]
看了一下你刚才的clean up，有几个问题
歌曲错误不仅仅是404，500、502、403各种状态码都有
有的歌曲在获得播放链接时会跳转，需要把302也当作200的一种
还有的歌曲客户端会分段下载，需要处理206（partial content）
建议可以直接statuscode/100，2和3就当作成功，4和5当作失败

还有一个是我当时写的时候的问题，有的歌曲url并不是.mp3结尾的，末尾还跟着?xxx=xxx的参数。需要把url.endwith换成url.contains

**Comment by EraserKing** [comment]
Thanks. I didn't expect so many exceptions. I'll modify it.

**Comment by EraserKing** [comment]
I forgot to mention it yesterday... It's modifed as well, all requests with response code >= 400 are treated as failures.

**Pull request merged**